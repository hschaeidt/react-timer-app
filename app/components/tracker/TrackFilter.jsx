import React, { Component, PropTypes } from 'react';

export default class TrackFilter extends Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    const showCompleted = this.showCompleted.checked;

    this.props.onFilter(showCompleted);
  }

  render() {
    return (
      <div>
        <label htmlFor="input-show-completed">
          <input
            id="input-show-completed"
            type="checkbox"
            onChange={this.handleChange}
            ref={(ref) => { this.showCompleted = ref; }}
          />
          Show completed days
        </label>
      </div>
    );
  }
}

TrackFilter.propTypes = {
  onFilter: PropTypes.func.isRequired,
};
