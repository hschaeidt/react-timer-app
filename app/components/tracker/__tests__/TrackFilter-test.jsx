import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import TrackFilter from '../TrackFilter';

describe('<TrackFilter />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<TrackFilter onFilter={jest.fn()} />);
    expect(tree).toMatchSnapshot();
  });

  it('should call onFilter on change', () => {
    const spy = jest.fn();
    const wrapper = mount(<TrackFilter onFilter={spy} />);

    wrapper.find('#input-show-completed').first().simulate('change', { target: { checked: true } });

    expect(spy).toHaveBeenCalled();
  });
});
