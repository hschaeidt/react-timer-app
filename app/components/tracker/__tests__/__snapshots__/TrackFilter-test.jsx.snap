exports[`<TrackFilter /> renders correctly 1`] = `
<div>
  <label
    htmlFor="input-show-completed">
    <input
      id="input-show-completed"
      onChange={[Function]}
      type="checkbox" />
    Show completed days
  </label>
</div>
`;
